cmd/sops/subcommand/filestatus/filestatus.go

```diff
@@ -0,0 +1,47 @@
+package filestatus
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"go.mozilla.org/sops"
+	"go.mozilla.org/sops/cmd/sops/codes"
+	"go.mozilla.org/sops/cmd/sops/common"
+)
+
+// Opts rapresent input option for FileStatus
+type Opts struct {
+	InputStore sops.Store
+	// OutputStore       sops.Store
+	InputPath string
+	// KeyServices       []keyservice.KeyServiceClient
+	// KeyGroups         []sops.KeyGroup
+	// GroupThreshold    int
+}
+
+// Status rapresent file status
+type Status struct {
+	Encrypted bool
+}
+
+func FileStatus(opts Opts) (Status, error) {
+	fileBytes, err := ioutil.ReadFile(opts.InputPath)
+	if err != nil {
+		return Status{}, common.NewExitError(fmt.Sprintf("Error reading file: %s", err), codes.CouldNotReadInputFile)
+	}
+	branches, err := opts.InputStore.LoadPlainFile(fileBytes)
+	if err != nil {
+		return Status{}, common.NewExitError(fmt.Sprintf("Error unmarshalling file: %s", err), codes.CouldNotReadInputFile)
+	}
+	hasMetadata := checkMetadata((branches[0]))
+	return Status{Encrypted: hasMetadata}, nil
+}
+
+func checkMetadata(branch sops.TreeBranch) bool {
+	for _, b := range branch {
+		if b.Key == "sops" {
+			return true
```

Comment:
isn't that a bit weak as a test? I'd suggest at least verifying that a `mac` is set and one key exists under the various sections.

Reply:
It's indeed a bit weak. It's what we do elsewhere as an easy test to check whether the file is encrypted, but maybe as you say we should have a stricter test here.

Reply:
May I propose to go with this version for consistency, and work on a more complex test (that we can discuss more) on a separate one? I'm willing to work on that.
We can also extend a bit the scope and extract the check so both places can use the same code.